7_react/react1/week2/todo-app/src/App.js

```diff
@@ -0,0 +1,36 @@
+// App.js
+import React, { useState } from 'react';
+import TodoList from './TodoList';
+
+function App() {
+  const [todos, setTodos] = useState([
+    { id: 1, description: 'Get out of bed', done: false },
+    { id: 2, description: 'Brush teeth', done: false },
+    { id: 3, description: 'Eat breakfast', done: false },
+    
+  ]);
+
+
+  const addRandomTodo = () => {
+    const id = todos.length + 1;
+    const description = `Random todo ${id}`;
+    setTodos([...todos, { id, description, done: false }]);
```

Comment:
On the surface all works as expected, but what happens if you remove and add some items? 

calculating the id based on the number of todo items may fail in some cases.

I.e. if you remove the first item and add a random item then you will end up todos with duplicate ids:
``` js
[
    { id: 2, description: 'Brush teeth', done: false },
    { id: 3, description: 'Eat breakfast', done: false },
    { id: 3, description: 'Random todo 3', done: false },
]
```
Try to figure out how you can find the highest id in the array and then add 1 to that instead.